numba_rvsdg/core/datastructures/scfg.py

```diff
@@ -845,14 +942,46 @@ def to_dict(self):
         graph_dict: Dict[Dict[...]]
             A dictionary representing the SCFG.
         """
-        scfg_graph = self.graph
-        graph_dict = {}
-        for key, value in scfg_graph.items():
-            curr_dict = {}
-            curr_dict["jt"] = [i for i in value._jump_targets]
-            if value.backedges:
-                curr_dict["be"] = [i for i in value.backedges]
-            graph_dict[key] = curr_dict
+        blocks = {}
+        edges = {}
+        backedges = {}
+
+        def reverse_lookup(value):
+            for k, v in block_type_names.items():
+                if v == value:
+                    return k
+            else:
+                raise TypeError("Block type not found.")
+
+        for key, value in self:
```

Comment:
Can this whole section use `isinstance` instead of string comparison?

Reply:
Resolved in [462c716](https://github.com/numba/numba-rvsdg/pull/50/commits/462c716676d5961ed48dffa0502c7c4f2e8e7878)